spec/plugins/mosaic/routes/searchMosaics.yml

```diff
@@ -0,0 +1,20 @@
+tags:
+  - Mosaic routes
+summary: Get mosaics
+description: Gets an array of mosaics.
+operationId: searchMosaics
+parameters:
+  - $ref: "../../../parameters/query/ownerAddress.yml"
```

Comment:
> can ownerAddress be a plural to support multiple owners?
> The wallet uses the load by multiple owner feature to avoid multiple requests. if you want by just own owner, you can query with a singleton [address]

@dgarcia360

Reply:
When defining the new endpoints, we couldn't think of a clear use case to request the mosaics created by different accounts at the same time. 

@fboucquez Could you please share a link to file where the wallet uses `POST /account/mosaics` to re-evaluate if the parameter ``ownerAddress`` should support multiple addresses? Thanks!

Reply:
https://github.com/nemfoundation/symbol-desktop-wallet/blob/master/src/services/MosaicService.ts#L109

The wallet handles multiple accounts in a profile. The wallet is prefetching and caching all of them for 1) offline access, 2) better user feedback 3) fewer rest requests that may cause too many requests.

Reply:
i don't care whether this endpoint stays or not, but it definitely is a weird endpoint

if the reason for its existence is this use by the wallet, it looks like a greedy client, fetching "everything" just in case - i also think getting a `too many requests` from just 5ish(?) queries instead of 1 is also something that should not determine how the API is structured... instead probably a less restrictive policy on the deploy is to be considered (it's not like the client is firing hundreds of requests)

for this extremely specific need of the client, a good approach would be to deliver a set of user data all at once (in any other session oriented API this'd be the user data required for the application to work), but that would come from a backend that serves a specific client, which isn't the case here

the client was designed to work offline, disregarding how the interface is built/works, so it's natural that we have those kinds of issues now... but i still think it'd be more natural to relax throttling config rather than requiring specific endpoints

Reply:
The wallet can live with calling rest multiple times, but if you can improve the usability of the mosaic search endpoint by replacing a `=` with `in` style query, maybe it worth the work? If the backed cost is N, let's keep it one then. 
This could be similar to the transaction types multi-value param in the search transaction endpoint.

Reply:
we had this conversation before and the reasons are exactly the same:
- it has never mattered from our perspective whether something is bigger or smaller effort to implement when deciding how to improve the project
- we don't see serving the specific needs of a single client to be the same as improving usability, and even more so when there is no reason to do so: the client can achieve the same results, with or without this endpoint, and not only that, but the client has extra work to do with this endpoint because since it is not something that makes sense to begin with, they still need to group by account at client side most likely (which sounds like more trouble than firing multiple requests), so to my eyes it's just a patch to bypass a potential `too many requests` error from a too restrictive server config, rather than adding usability

Reply:
The wallet is just one need, it may be other. The getMosaicByOnwers was already in the spec, we are reducing functionality. That method can be superseded by the mosaic search if it accepts multiple owners.
Creating a list of addresses in memory is less trouble and more I/O efficient that firing multiple rest calls and then aggregate the async results together. And if the user really wants to call one by one, he can by sending multiple queries with one address each. It's not just a patch, it is saving I/O and it makes the endpoint more powerful (if the backend cost is not too high needing some complicated join queries)

Reply:
- an endpoint where you add a list of keys, and it returns, in a single list, the mixed results that requires postprocessing by the client, however you look at this this endpoint is not well designed - for it to be really useful it'd have to at least return the results with grouped values by the owners (and this would be a specificity and not very RESTful)
- this endpoint was added [6 months ago](https://github.com/nemtech/catapult-rest/issues/11) and only to "copy" another endpoint, no other reasons that i can see
- i think there's no point discussing performance at this level because as i mentioned, that one out of 2000 clients requires to do 5 requests instead of 1 is insignificant
- we are not losing functionality
- i was not defending to remove this endpoint, in fact, i noticed it disappeared, and i asked here if this was intended - david decided to remove it because it seemed a rather redundant  / out of place endpoint, yet more proof that this is not fitting for this API

Reply:
just let @dgarcia360 make a decision, as i said, we don't mind whether this stays or not, but we still think it's a badly designed endpoint that has no place or added value to the API

Reply:
I'd prefer to keep IN queries to the bare minimum so that we don't leave open the possibility to apply the same solution to almost every other query param without a clear usage.

In the wallet case, I believe the previous POST endpoint was used because it was available, but not because it was essential or helped to enhance the developer experience. 

When an account (or the mosaic owned screen ) is loaded, the wallet has to fetch also all the mosaics owned by the account to determine if there is some new mosaic or a definition that has changed (assuming there is internet connection). At this point, the latest mosaics owned could be stored locally in case they are needed for offline usage.

Reply:
Fair enough. Please remove the original endpoint (as you still need to limit/paginate if there are 1000s of mosaics) and keep the search owner singular. The wallet can do multiple requests.